src/js/views/panel_view.js

```diff
@@ -196,23 +196,42 @@
                     var last_idx = 0;
                     for (const match of matches) {// Loops on the match to replace in the ljson.text the expression by their values
                         var new_text = new_text + ljson.text.slice(last_idx, match.index);
-                        expr = match[0].slice(1,-1).split(".");
-                        var label_value = ""
-                        if (['time', 't'].includes(expr[0])) {
-                            label_value = self.model.get_time_label_text(expr[1] ? expr[1] : "index");
-                        } else if (['image', 'dataset'].includes(expr[0])){
-                            label_value = self.model.get_name_label_text(expr[0], expr[1] ? expr[1] : "name");
+
+                        // Label parsing in three steps:
+                        //   - split label type.format from other parameters (;)
+                        //   - split label type and format (.)
+                        //   - grab other parameters (key=value)
+                        expr = match[0].slice(1,-1).split(";");
+                        prop_nf = expr[0].trim().split(".")
+                        param_dict = {};
+                        expr.slice(1).forEach(function(value) {
+                            kv = value.split("=");
+                            if (kv.length > 1) {
+                                param_dict[kv[0].trim()] = parseInt(kv[1].trim());
+                            }
+                        });
+
+                        var label_value = "";
+                        if (['time', 't'].includes(prop_nf[0])) {
+                            format = prop_nf[1] ? prop_nf[1] : "index";
+                            precision = param_dict["precision"] !== undefined ? param_dict["precision"] : 0; // decimal places default to 0
+                            label_value = self.model.get_time_label_text(format, param_dict["offset"], precision);
+                        } else if (['image', 'dataset'].includes(prop_nf[0])){
+                            format = prop_nf[1] ? prop_nf[1] : "name";
+                            label_value = self.model.get_name_label_text(prop_nf[0], format);
```

Comment:
Need to check that all these variable declarations have `var` too, or they end up as global variables.